android/src/main/java/io/ably/flutter/plugin/AblyFlutterPlugin.java

```diff
@@ -65,4 +82,55 @@ public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
     private static MethodCodec createCodec() {
         return new StandardMethodCodec(new AblyMessageCodec());
     }
+
+    @Override
+    public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
+        isActivityRunning = true;
+        mainActivity = binding.getActivity();
+        binding.addOnNewIntentListener(this);
+        Intent intent = mainActivity.getIntent();
+        doIfIntentContainsRemoteMessage(intent, (message) -> {
+            // Only send the RemoteMessage to the dart side if it was actually a RemoteMessage.
+            methodCallHandler.setRemoteMessageFromUserTapLaunchesApp(message);
+        });
+    }
+
+    @Override
+    public void onDetachedFromActivityForConfigChanges() {
+    }
+
+    @Override
+    public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBinding binding) {
+        mainActivity = binding.getActivity();
+        binding.addOnNewIntentListener(this);
+    }
+
+    // This method does not get called when the app goes into the background
+    @Override
+    public void onDetachedFromActivity() {
+        isActivityRunning = false;
+    }
+
+    @Override
+    public boolean onNewIntent(Intent intent) {
+        mainActivity.setIntent(intent);
+        doIfIntentContainsRemoteMessage(intent, (message) -> {
+            methodChannel.invokeMethod(PlatformConstants.PlatformMethod.pushOnNotificationTap, message);
+        });
+        return false;
+    }
+
+    private Boolean doIfIntentContainsRemoteMessage(Intent intent, CompletionHandlerWithRemoteMessage completionHandler) {
+        Bundle extras = intent.getExtras();
+        if (extras == null) {
+            return false;
+        }
+        RemoteMessage message = new RemoteMessage(intent.getExtras());
+        // TODO should i sent a RemoteMessage if only notificaiton is populated
```

Comment:
I've manually unindented those lines, since autoformatting the file changed every line in the file 😅

The todo was actually still relevant, but I've fixed it now. I now only send a message to the dart side if the dictionary is not empty. Notifications only messages are not sent to the dart side, to be consistent with iOS. 

55f111fd8fa15a4307b2ed26b494311bea508137